.github/workflows/ci.yml

```
@@ -0,0 +1,102 @@
+name: CI
+on:
+  push:
+    branches:
+      - master
+      - release-[8-9].*
+      - release-1[0-9].*
+  pull_request:
+      branches-ignore:
+        - release-[0-7].*
+  pull_request_target:
+      branches-ignore:
+        - release-[0-7].*
+  workflow_dispatch:
+env:
+  DOTNET_NOLOGO: true
+jobs:
+  build:
+    if:
+      (github.event_name == 'pull_request_target' && github.event.pull_request.user.login == 'dependabot[bot]') ||
+      (github.event_name == 'pull_request' && github.event.pull_request.user.login != 'dependabot[bot]') ||
+      github.event_name == 'push' || github.event_name == 'workflow_dispatch'
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: windows-2019
+            name: Windows-Sql
+          - os: ubuntu-20.04
+            name: Linux
+          - os: [self-hosted, windows, oracle]
+            name: Windows-Oracle
+      fail-fast: false
+    steps:
+      - name: Check for secrets
+        env:
+          SECRETS_AVAILABLE: ${{ secrets.SECRETS_AVAILABLE }}
+        shell: pwsh
+        run: exit $(If ($env:SECRETS_AVAILABLE -eq 'true') { 0 } Else { 1 })
+      - name: Checkout
+        if: github.event_name != 'pull_request_target'
+        uses: actions/checkout@v2.4.0
+        with:
+          fetch-depth: 0
+      - name: Checkout for Dependabot
+        if: github.event_name == 'pull_request_target'
+        uses: actions/checkout@v2.4.0
+        with:
+          ref: "refs/pull/${{ github.event.number }}/merge"
+          fetch-depth: 0   
+      - name: Setup .NET 6
+        uses: actions/setup-dotnet@v1.9.0
+        with:
+          dotnet-version: 6.0.x
+      - name: Setup .NET 5 
+        uses: actions/setup-dotnet@v1.9.0
+        with:
+          dotnet-version: 5.0.x          
+      - name: Setup .NET Core 3.1 
+        uses: actions/setup-dotnet@v1.9.0
+        with:
+          dotnet-version: 3.1.x   
```

Since you don't need net6.0 (yet) I wonder if switching to this would have any effect on the build problem you're getting on the self-hosted agent: (This combined versions stuff is possible since setup-dotnet 1.9.0)
```suggestion
      - name: Setup .NET SDK
        uses: actions/setup-dotnet@v1.9.0
        with:
          dotnet-version: |
            5.0.x
            3.1.x
```